packages/carequality-sdk/src/client/__tests__/carequality.test.ts

```diff
@@ -0,0 +1,179 @@
+  if (apiMode === "dev") return APIMode.dev;
+  throw new Error(`Invalid API_MODE: ${apiMode}`);
+}
+
+function makeCarequalityManagementAPI(): CarequalityManagementApi | undefined {
+  const apiKey = getEnvVar("MANAGEMENT_API_KEY");
+  const orgCert = getEnvVar("CQ_ORG_CERTIFICATE");
+  const rsaPrivateKey = getEnvVar("CQ_ORG_PRIVATE_KEY");
+  const rsaPrivateKeyPassword = getEnvVar("CQ_ORG_PRIVATE_KEY_PASSWORD");
+  if (!apiKey || !orgCert || !rsaPrivateKey || !rsaPrivateKeyPassword) return undefined;
+  const apiMode = getApiMode();
+  return new CarequalityManagementApiFhir({
+    apiKey,
+    apiMode,
+    orgCert,
+    rsaPrivateKey,
+    rsaPrivateKeyPassword,
+  });
+}
+
+type OrgWithId = Organization & Required<Pick<Organization, "id">>;
+
+// TODO Need to setup CICD to support running these automatically
+describe("CarequalityManagementAPIImpl", () => {
+  if (getApiMode() === APIMode.production) {
+    it.skip("Can't be run in production");
+    return;
+  }
+  const api = makeCarequalityManagementAPI();
+  if (!api) {
+    console.log("Skipping tests because the CQ API is not configured (missing env vars)");
+    it.skip("skipping tests", () => {});
+    return;
+  }
+  const managementOid = getEnvVarOrFail("MANAGEMENT_ORG_OID");
+
+  let oid: string | undefined = undefined;
+  async function getOid(): Promise<string> {
+    if (oid) return oid;
+    for (let i = 0; i < 10; i++) {
+      oid = managementOid + ".5." + faker.number.int({ min: 1, max: 999_999_999 });
+      const org = await api!.getOrganization(oid);
+      if (!org) return oid;
+    }
+    throw new Error("Failed to get a unique OID");
+  }
+
+  async function makeOrganization(): Promise<OrgWithId> {
+    const oid = await getOid();
+    const orgFhir = getOrganizationFhirTemplate({
+      oid,
+      name: faker.company.name(),
+      addressLine1: faker.location.streetAddress(),
+      city: faker.location.city(),
+      state: faker.location.state(),
+      postalCode: faker.location.zipCode(),
+      lat: faker.location.latitude().toString(),
+      lon: faker.location.longitude().toString(),
+      contactName: faker.person.fullName(),
+      phone: faker.phone.number(),
+      email: faker.internet.email(),
+      role: "Connection",
+      active: false,
+      parentOrgOid: managementOid,
+    });
+    return orgFhir as OrgWithId;
+  }
+
+  let orgCreate: OrgWithId | undefined;
+  let org: OrgWithId | undefined;
+
+  // beforeAll(async () => {
+
+  // });
+
+  // TODO we should have a beforeAll()/afterAll() to setup/teardown the test data
+  // so the result of the tests is more deterministic.
+
+  describe("createOrganization", () => {
+    it("registers a new organization", async () => {
+      orgCreate = await makeOrganization();
+      const org = await api.registerOrganization(orgCreate);
+      // TODO remove this
+      // TODO remove this
+      // TODO remove this
+      console.log(`resp registerOrganization: `, JSON.stringify(org, null, 2));
```

Comment:
think u forgot about this

Reply:
Good catch, but it's not part of the final PR version - maybe an intermediary commit? Those get outdated, sorry - can use as a reference, but not for the respective code's final version.